pkg/shp/cmd/auth.go

```diff
@@ -0,0 +1,10 @@
+package cmd
```

Comment:
In my other projects, I usually tried to place those "underscore" imports into the same package and source file where I implemented the Kubernetes cluster authentication and clientset retrieval. However, I could not really figure out a good place here as the clientset retrieval calls might happen in different places, depending on what you run. So I figured a explicitly named file might do the trick here.

From what I can see with the `dependencymagnet`, we have a slightly different use case here as we always require these imports so that their respective `init` functions are called. I guess this is what we really need here.

Anyway, happy to refactor it if we have a better idea on how we want to name things and whether there is a better style for solving this kind of requirement.

Reply:
As I recall `dependencymagnet` is an "always" thing ... I could be missing something, but I don't *think* I am :-).  What made you think precisely that it is insufficient @HeavyWombat ?  Did you try it and something was amiss ?

Let's wait a little to see if others like @adambkaplan @otaviof chime in once they return / catch up from being out.  Otherwise, let's definitely discuss this during the Monday community meeting.  Even if it is a "we do what you have know, but agree on a different approach to be handled as a follow up" during the meeting, I won't hold on this past Monday.

Reply:
I think I was concerned with the build tags: `// +build tools`. It looked like the package was "only" for dependencies that were kind of tools and only required for build time, not runtime.

Reply:
yes the build tool scenario is another reason to employ

went back and revisited openshift/api and yes, while the package alone handles the vendoring, there are no imports of it, and yes with your case an import of the package is needed for the init functions, and you get that import because of the package you placed this in is imported for other function related here

that said, I still find the naming here a bit lacking .... if not misleading, definitely not fully informative.

I think a tweak on the dependencymagnet approach could be better .... changing the name of this file to `dependencymagnet.go` would be more descriptive of its intent.  Also, it is generic such that if we had future dependencies of this nature, other than "auth", we could add them here, without having to worry about changing from auth.go to something else.

If you are amenable to my idea as is, great, go for it.  But I certainly understand if you'd like to get a broader consensus amongst the community.

If so, let's bring it up on Monday's call.